src/arch/x86_64/kernel/percore.rs

```diff
@@ -115,14 +106,15 @@ impl Is32BitVariable for u32 {}
 impl<T: Is32BitVariable> PerCoreVariableMethods<T> for PerCoreVariable<T> {
 	#[inline]
 	unsafe fn get(&self) -> T {
-		let value: T;
-		llvm_asm!("movl %gs:($1), $0" : "=r"(value) : "r"(self.offset()) :: "volatile");
-		value
+		let value: u32;
```

Comment:
is `u32` here on purpose?

Reply:
Yes, as we want to explicitly use 32-bit memory moves here.  

https://github.com/hermitcore/libhermit-rs/blob/d7ea077c0f9bc4fb7412aa3e7bb67c68a41c7267/src/arch/x86_64/kernel/percore.rs#L104-L105